doc/source/serve/dev-workflow.md

```diff
@@ -0,0 +1,109 @@
+# (ServeController pid=9865) INFO 2022-08-11 11:31:55,373 controller 9865 deployment_state.py:1232 - Adding 1 replicas to deployment 'Doubler'.
+# (ServeController pid=9865) INFO 2022-08-11 11:31:55,389 controller 9865 deployment_state.py:1232 - Adding 1 replicas to deployment 'HelloDeployment'.
+# (HTTPProxyActor pid=9872) INFO:     Started server process [9872]
+# 2022-08-11 11:31:57,383 SUCC scripts.py:315 -- Deployed successfully.
+```
+
+The `serve run` command blocks the terminal and can be canceled with Ctrl-C.
+
+Now that Serve is running, we can send HTTP requests to the application.
+For simplicity, we'll just use the `curl` command to send requests from another terminal.
+
+```bash
+curl -X PUT "http://localhost:8000/?name=Ray"
+# Hello, Ray! Hello, Ray!
+```
+
+After you're done testing, you can shut down Ray Serve by interrupting the `serve run` command (e.g., with Ctrl-C):
+
+```console
+^C2022-08-11 11:47:19,829       INFO scripts.py:323 -- Got KeyboardInterrupt, shutting down...
+(ServeController pid=9865) INFO 2022-08-11 11:47:19,926 controller 9865 deployment_state.py:1257 - Removing 1 replicas from deployment 'Doubler'.
+(ServeController pid=9865) INFO 2022-08-11 11:47:19,929 controller 9865 deployment_state.py:1257 - Removing 1 replicas from deployment 'HelloDeployment'.
+```
+
+## Testing on a remote cluster
+
+To test on a remote cluster, you'll use `serve run` again, but this time you'll pass in an `--address` argument to specify the address of the Ray cluster to connect to.  For remote clusters, this address has the form `ray://<head-node-ip-address>:10001`; see [Ray Client](ray-client-under-construction) for more information.
+
+When making the transition from your local machine to a remote cluster, you'll need to make sure your cluster has a similar environment to your local machine--files, environment variables, and Python packages, for example.  During development, you can use {ref}`Runtime Environments<runtime-environments>` to manage these dependencies.
+
+Let's see a simple example. Run the following command on your local machine, with your remote cluster head node IP address substituted for `<head-node-ip-address>` in the command:
+
+```bash
+serve run  --address=ray://<head-node-ip-address>:10001 --working_dir="./project/src" local_dev:HelloDeployment
+```
+
+This will connect to the remote cluster via Ray Client, upload the `working_dir` directory, and run your serve application.  Here, the local directory specified by `working_dir` must contain `local_dev.py` so that it can be uploaded to the cluster and imported by Ray Serve.
+
+Once this is up and running, we can send requests to the application:
+
+```bash
+curl -X PUT http://<head-node-ip-address>:8000/?name=Ray
+# Hello, Ray! Hello, Ray!
+```
+
+If your runtime environment contains fields other than `working_dir`, use the `--runtime-env-json` argument:
```

Comment:
```suggestion
When making the transition from your local machine to a remote cluster, you'll need to make sure your cluster has a similar environment to your local machine--files, environment variables, and Python packages, for example.

Let's see a simple example that just packages the code. Run the following command on your local machine, with your remote cluster head node IP address substituted for `<head-node-ip-address>` in the command:

```bash
serve run  --address=ray://<head-node-ip-address>:10001 --working_dir="<directory containing local_dev.py>" local_dev:HelloDeployment
```
This will connect to the remote cluster via Ray Client, upload the directory specified by `--working_dir`, and run your Serve application in this directory. We can make `local_dev.py` importable by Ray Serve by making sure that it is contained in the directory passed to `--working-dir`.

Once this is up and running, we can send requests to the application:

```bash
curl -X PUT http://<head-node-ip-address>:8000/?name=Ray
# Hello, Ray! Hello, Ray!
```

For more complex dependencies, including files outside the working directory, environment variables, and Python packages, you can use {ref}`Runtime Environments<runtime-environments>` to manage these dependencies. Here is an example using the `--runtime-env-json` argument:
```

Reply:
Oh no, I think my suggestion got mis-formatted by GitHub :dagger: 

But basically I was trying to:
1. move the mention of runtime envs closer to where it's actually used in a command (for a first-time user, I don't think it will be obvious that --working-dir is just syntactic sugar for a runtime env)
2. explain semantics of `--working-dir`

Reply:
Makes sense!